src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1042,45 +1042,45 @@ LedgerMaster::checkAccept(std::shared_ptr<Ledger const> const& ledger)
 
     if (ledger->seq() % 256 == 0)
     {
-        /*
-         * Check if the majority of validators run a higher version rippled
-         * software. If so print a warning.
-         *
-         * Once the HardenedValidations amendment is enabled, validators include
-         * their rippled software version in the validation messages of every
-         * (flag - 1) ledger. We wait for one ledger time before checking the
-         * version information to accumulate more validation messages.
-         */
-        auto const& patentHash = ledger->info().parentHash;
-        auto const parentLedger = getLedgerByHash(patentHash);
+        // Check if the majority of validators run a higher version rippled
+        // software. If so print a warning.
+        //
+        // Once the HardenedValidations amendment is enabled, validators include
+        // their rippled software version in the validation messages of every
+        // (flag - 1) ledger. We wait for one ledger time before checking the
+        // version information to accumulate more validation messages.
+
+        auto const& parentHash = ledger->info().parentHash;
+        auto const parentLedger = getLedgerByHash(parentHash);
         if (parentLedger &&
             parentLedger->rules().enabled(featureHardenedValidations))
         {
-            auto const vals =
-                app_.getValidations().getTrustedForLedger(patentHash);
-            auto higherVersionCount = std::count_if(
-                vals.begin(), vals.end(), [](auto const& v) -> bool {
-                    if (v->isFieldPresent(sfServerVersion))
-                        return BuildInfo::localVersionLow(
-                            v->getFieldU64(sfServerVersion));
-                    else
-                        return false;
-                });
-            auto const threshold = static_cast<std::size_t>(std::ceil(
-                app_.validators().getQuorumKeys().second.size() *
-                BuildInfo::versionUpgradeWarningThreshold));
-            if (higherVersionCount >= threshold)
+            // To throttle the warning messages, instead of printing a warning
+            // every flag ledger, we print every week.
+            static TimeKeeper::time_point lastTime{};
```

Comment:
This should be a member of `LedgerMaster` instead of a scope-level `static`.

Reply:
Thanks, changed.